### PR TITLE
[Model] Add Unlimited-OCR support

### PR DESCRIPTION
**Pull request by calyoung80**
Add vLLM support for Unlimited-OCR.

  ## Purpose

  This PR adds Unlimited-OCR as a supported multimodal generation model in vLLM.

  Main changes:
  - Add `UnlimitedOCRConfig`.
  - Add `UnlimitedOCRForCausalLM` and processing logic based on the DeepSeek OCR family.
  - Register Unlimited-OCR in the model registry and model test registry.
  - Fix tokenizer loading for `model_type=unlimited-ocr` by forcing the generic fast tokenizer backend when the hub tokenizer class is incorrect.
  - Register the chat template fallback for `unlimited-ocr`, using the same DeepSeek OCR template path so BOS handling matches the reference path.
  - Add focused model tests for config loading, registry import, image token accounting, and tokenizer decoding.

  AI assistance disclosure:
  - This PR was prepared with AI assistance. I reviewed the generated changes and validated the behavior with the tests below.

  ## Test Plan

  ```bash
  SKIP=mypy-local python -m pre_commit run --all-files --show-diff-on-failure

  python -m pytest tests/models/test_unlimited_ocr.py -q --confcutdir=tests/models

  python -m pytest tests/models/test_registry.py -q \
    -k "unlimited or Unlimited" \
    --confcutdir=tests/models

  Manual E2E validation on Ascend/NPU:

  vllm serve /mnt/share/Unlimited-OCR \
    --served-model-name Unlimited-OCR \
    --tensor-parallel-size 1 \
    --max-model-len 8192 \
    --mm-processor-cache-gb 0 \
    --port 8000 \
    --enforce-eager \
    --logits-processors vllm.model_executor.models.deepseek_ocr:NGramPerReqLogitsProcessor

  Then send an OpenAI-compatible chat completion request with a receipt image and prompt:

  <image>document parsing.

  ## Test Result

  SKIP=mypy-local python -m pre_commit run --all-files --show-diff-on-failure

  Result:

  Passed:
  - ruff check
  - ruff format
  - typos
  - clang-format
  - markdownlint-cli2
  - actionlint
  - pip-compile
  - pip-compile-rocm
  - pip-compile-xpu
  - pip-compile-docs
  - nightly torch requirements formatter
  - shellcheck
  - png-lint
  - SPDX header check
  - root lazy imports check
  - filename check
  - Dockerfile dependency graph check
  - forbidden imports check
  - torch.cuda API check
  - config validation
  - docker/versions validation
  - attention backend docs check
  - boolean context manager check

  mypy-local was skipped for the full pre-commit run because it fails in this local Ascend environment on unrelated existing files outside the Unlimited-OCR changes.

  Model tests:

  python -m pytest tests/models/test_unlimited_ocr.py -q --confcutdir=tests/models
  4 passed, 16 warnings

  python -m pytest tests/models/test_registry.py -q -k "unlimited or Unlimited" --confcutdir=tests/models
  1 passed, 367 deselected, 16 warnings

  Manual E2E result:

  /v1/models returns Unlimited-OCR
  chat completion returns non-empty OCR output
  prompt_tokens = 277
  completion_tokens = 305
  result_len = 735

  The OCR output no longer contains ByteLevel decoder artifacts such as Ġ, Ċ, or mojibake fragments, and the receipt output is aligned with the SGLang/reference path.

  ———
  <details>
  <summary> Essential Elements of an Effective PR Description Checklist </summary>

  - [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
  - [x] The test plan, such as providing test command.
  - [x] The test results, such as pasting the results comparison before and after, or e2e results
  - [ ] (Optional) The necessary documentation update, such as updating supported_models.md and examples for a new model.
  </details>

  BEFORE SUBMITTING, PLEASE READ https://docs.vllm.ai/en/latest/contributing (https://docs.vllm.ai/en/latest/contributing) (anything written below this line will be removed by  GitHub Actions)

### CODE REVIEW

**Comment by calyoung80** [comment]
Hi @DarkLight1337 @ywang96 @AndreasKaratzas @njhill, this is my first vLLM PR, so CI is blocked by:

  > PR must have the `verified` or `ready` label or the author must have at least 4 merged PRs.

  Could you please review and add the `ready` or `verified` label to trigger CI?

  I have run the relevant checks locally:
  - targeted pre-commit on changed files: passed
  - `python -m pytest tests/models/test_unlimited_ocr.py -q --confcutdir=tests/models`: 4 passed
  - `python -m pytest tests/models/test_registry.py -q -k "unlimited or Unlimited" --confcutdir=tests/models`: 1 passed
  - full pre-commit passed with `SKIP=test-nonroot-entrypoint`; the skipped hook fails in my local container environment and is unrelated to this PR.

**Comment by noooop** [comment]
Duplicate PR
